tests/unit/test_slash_command_resolver.py

```
@@ -0,0 +1,225 @@
+        return list(_CANNED_VECTORS[text])
+    # Unknown strings get an orthogonal-ish vector so they never collide.
+    return [0.0, 0.0, 0.0]
+
+
+class StubEmbeddingStore:
+    """In-memory deterministic EmbeddingStoreBackend stub."""
+
+    def __init__(self) -> None:
+        self.rows: Dict[str, Dict[str, Any]] = {}
+
+    async def add(self, id: str, embedding: List[float], metadata: Dict[str, Any]) -> None:
```

Ruff (0.15.12)

[error] Function argument `id` is shadowing a Python builtin

(A002)